__tests__/replace-libraries-ngpackagr-builder.test.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  angular10_1AndUpNgPackagrBuilder,
+  preAngular10_1NgPackagrBuilder,
+  replaceLibrariesNgPackagrBuilder
+} from '../src/replace-libraries-ngpackagr-builder';
+import {lumberjackMixedAngularJson} from './angular-json/lumberjack-mixed-angular-json';
+import {lumberjack10_1AndUpNgPackagrBuilderAngularJson} from './angular-json/lumberjack-angular10_1-and-up-ngpackagr-builder-angular-json';
+import {lumberjackPreAngular10_1NgPackagrBuilderAngularJson} from './angular-json/lumberjack-pre-angular10_1-ngpackagr-builder-angular-json';
+import {regularPreAngular10_1NgPackagrBuilderAngularJson} from './angular-json/regular-pre-angular10_1-ngpackagr-builder-angular-json';
+import {regularAngular10_1AndUpNgPackagrBuilderAngularJson} from './angular-json/regular-angular10_1-and-up-ngpackagr-builder-angular-json';
+import {AngularVersion} from '../src/types/angular-version';
+
+const preAngular10_1Versions: AngularVersion[] = [
+  '6.0.x',
+  '6.1.x',
+  '7.0.x',
+  '7.1.x',
+  '7.2.x',
+  '8.0.x',
+  '8.1.x',
+  '8.2.x',
+  '9.0.x',
+  '9.1.x',
+  '10.0.x'
+];
+
+const angular10_1AndUpVersions: AngularVersion[] = [
+  '10.1.x',
+  '10.2.x',
+  '11.0.x',
+  '11.1.x',
+  '11.2.x'
+];
+
+describe(replaceLibrariesNgPackagrBuilder.name, () => {
+  preAngular10_1Versions.forEach(version => {
+    test(`when using the version ${version} it should replace all libraries ng-packagr builder with ${preAngular10_1NgPackagrBuilder}`, () => {
+      const actualLumberjackAngularJson = replaceLibrariesNgPackagrBuilder(
+        version,
+        lumberjackMixedAngularJson
+      );
+      const actualRegularAngularJson = replaceLibrariesNgPackagrBuilder(
+        version,
+        regularAngular10_1AndUpNgPackagrBuilderAngularJson
+      );
+
+      expect(actualLumberjackAngularJson).toEqual(
+        lumberjackPreAngular10_1NgPackagrBuilderAngularJson
+      );
+      expect(actualRegularAngularJson).toEqual(
+        regularPreAngular10_1NgPackagrBuilderAngularJson
+      );
+    });
+  });
+
+  angular10_1AndUpVersions.forEach(version => {
+    test(`when using the version ${version} it should replace all libraries ng-packagr builder with ${angular10_1AndUpNgPackagrBuilder}`, () => {
+      const actualLumberjackAngularJson = replaceLibrariesNgPackagrBuilder(
+        version,
+        lumberjackMixedAngularJson
+      );
+      const actualRegularAngularJson = replaceLibrariesNgPackagrBuilder(
+        version,
+        regularPreAngular10_1NgPackagrBuilderAngularJson
+      );
+
+      expect(actualLumberjackAngularJson).toEqual(
+        lumberjack10_1AndUpNgPackagrBuilderAngularJson
+      );
+      expect(actualRegularAngularJson).toEqual(
+        regularAngular10_1AndUpNgPackagrBuilderAngularJson
+      );
+    });
+  });
+});
```

Comment:
>(Lars) What I mean is let the test show that the builder name is what's changed. This is not currently clear in the test body, only in the test description. Assert on the builder name of a library `build` architect target, not the entire workspace configuration.

>(Nacho) I know what you are saying but the reason I like to do it this way is that it gives me confidence that I'm not changing other target builders that I should not be changing.
Just asserting against the builder that should change is not covering that kind of errors. The solution could be checking builder by builder, I don't like that, but I would wait for your thoughts on this one.

suggestion: Add unit tests with a narrow scope like I suggested in addition to your tests with very broad scopes. It requires a lot of thinking and code navigation to figure out and verify what they are actually testing.